snapcraft/_store.py

```diff
@@ -321,6 +322,10 @@ def close_channels(
     ) -> Dict[str, Any]:
         return super().close_channels(snap_id=snap_id, channel_names=channel_names)
 
+    @_login_wrapper
```

Comment:
Look at the validatic/gated logic where a dance is made to re-request a macaroon if lacking a caveat
```
RuntimeError: Uh oh: {'error_list': [{'message': 'Permission "package_metrics" is required as a macaroon caveat.', 'code': 'macaroon-permission-required', 'extra': {'permission': 'package_metrics'}}]}
```